src/modules/campaign/ContentfulClient.spec.ts

```diff
@@ -1,87 +1,273 @@
+          client.fetchEntryAllLocales(mockSpace, mockEnvironment, mockId)
+        ).rejects.toThrow('Error fetching entry in all locales')
+      })
+    })
+  })
+
+  describe('fetchAsset', () => {
+    describe('when the request is successful', () => {
+      beforeEach(() => {
+        nock(CMS_URL)
+          .get(
+            `/spaces/${mockSpace}/environments/${mockEnvironment}/assets/${mockId}/`
+          )
+          .reply(200, marketplaceHomepageBannerAssets[0])
+      })
+
+      it('should fetch and transform an asset to the correct format', async () => {
+        const result = await client.fetchAsset(
           mockSpace,
           mockEnvironment,
-          mockId,
-          mockContentType,
-          mockToken
+          mockId
         )
 
-        expect(result).toEqual(mockResponse)
+        expect(result.fields).toHaveProperty('title.en-US')
+        expect(result.fields).toHaveProperty('description.en-US')
+        expect(result.fields).toHaveProperty('file.en-US')
       })
     })
 
     describe('when the request fails', () => {
       beforeEach(() => {
-        global.fetch = jest.fn().mockResolvedValue({
-          ok: false
-        })
+        nock(CMS_URL)
+          .get(
+            `/spaces/${mockSpace}/environments/${mockEnvironment}/assets/${mockId}/`
+          )
+          .replyWithError('Failed to fetch asset data')
       })
 
       it('should throw an error', async () => {
         await expect(
-          client.fetchEntry(
-            mockSpace,
-            mockEnvironment,
-            mockId,
-            mockContentType,
-            mockToken
-          )
-        ).rejects.toThrow('Failed to fetch entity data')
+          client.fetchAsset(mockSpace, mockEnvironment, mockId)
+        ).rejects.toThrow('Failed to fetch asset data')
       })
     })
   })
+
+  describe('fetchAssetsFromEntryFields', () => {
+    describe('when there are fields to get the assets from', () => {
+      it('should get all assets referenced in the entry fields', async () => {
+        global.fetch = jest.fn().mockResolvedValue({
+          ok: true,
+          json: () => Promise.resolve(marketplaceHomepageBannerAssets[0])
+        })
+
+        const result = await client.fetchAssetsFromEntryFields(
+          mockSpace,
+          mockEnvironment,
+          [mockFields]
+        )
+
+        expect(result).toHaveProperty(marketplaceHomepageBannerAssets[0].sys.id)
+        expect(Object.keys(result)).toHaveLength(1)
+      })
+    })
+
+    describe('when there are no entry fields to look assets from', () => {
+      it('should return empty object', async () => {
+        const result = await client.fetchAssetsFromEntryFields(
+          mockSpace,
+          mockEnvironment,
+          []
+        )
+        expect(result).toEqual({})
+      })
+    })
+  })
+
+  describe('fetchEntriesFromEntryFields', () => {
+    it('should fetch all entries referenced in the entry fields', async () => {
```

Comment:
This test does not match the testing standard. Would you mind doing the same that you did with the other tests here? The context must be described for each `it` and a `beforeEach` must be used to define the context.